tests/jballerina-unit-test/src/test/resources/test-src/javainterop/inferred_dependently_typed_func_signature_negative.bal

```diff
@@ -185,3 +185,15 @@ function testInvalidUsageWithCasts() {
     var a = <function (function, boolean)> getFunctionWithAnyFunctionParamType(function (function x, int y) { });
     BooleanStream|handle b = <BooleanStream> check funcReturningUnionWithBuiltInRefType(());
 }
+
+function fooFn1(typedesc td = <>) returns td[] = external;
+
+function fooFn2(typedesc<int|string> td1, typedesc td2 = <>) returns [td1, td2] = external;
+
+function fooFn3(typedesc t, typedesc<boolean> td = <>) returns int|t|td = external;
+
+function testInvalidUsageWithoutContextuallyExpectedTypeAndArgument() {
+    fooFn1();
+    fooFn2(int);
+    fooFn3(float);
```

Comment:
I know we already have tests, but for completeness shall we add a test with var also here?

Reply:
Added tests 905aecbe4006052e3602df03d55026660556f8b6